avocado/plugins/human.py

```diff
@@ -92,7 +96,20 @@ def end_test(self, result, state):
         duration = (" (%.2f s)" % state.get('time_elapsed', -1)
                     if status != "SKIP"
                     else "")
-        msg = self.get_colored_status(status, state.get("fail_reason", None))
+        if self.runner == 'nrunner':
+            if "name" in state:
+                name = state["name"]
+                uid = name.str_uid
+                name = name.name + name.str_variant
+            else:
+                name = "<unknown>"
+                uid = '?'
+
+            msg = self.get_colored_status(status, state.get("fail_reason", None))
+            LOG_UI.debug(' (%s/%s) %s:  ', uid, result.tests_total, name,
+                         extra={"skip_newline": True})
+        else:
+            msg = self.get_colored_status(status, state.get("fail_reason", None))
```

Comment:
This dichotomy of runners might influence multiple plugins and some of us have their own inheriting classes so if in the future the handling diverges too much requiring `if ... else` in all important plugins, this might as well be better handled with some data separation through multiple specialized classes.

Reply:
Absolutely... this is actually a violation of layers.  One plugin should be bothered by another.  But this is due to the fact that we don't have a nice presentation layer currently.

Now about the runner itself, I don't think there's any need to have multiple runners *after* LTS.  Our plan is to keep nrunner and drop the current one.

Reply:
I think the plan was to make the nrunner the default option and support the serial runner at least for some good amount of time. Dropping the (serial) runner completely will deprecate a lot of plugins out there including Avocado VT.

Reply:
The goal is to support the current (AKA serial) runner for a long time indeed, but within the LTS release/branch.  That's why this upcoming LTS release is rather important.

The reason for doing so is that there's consensus (AFAIK) that mandating that Avocado-VT works with older and newer Avocado is hurting the progress of both projects.  I believe Avocado-VT will not be hurt by sticking to 82.x (with all fixes it needs) until a more modern Avocado-VT can be ported into, say, Avocado 100.0.

Does it make sense?

Reply:
Hmmm I guess it does even though this will require a much bigger and more serious talk. Because as a matter of fact, migrating and/or updating Avocado VT would not be anything easily feasible. I do agree that indeed, it is often staying on the path of progress and has the additional weight of many parties that find it harder to update, be it due to resource constraints or even less activity. But it is also one of the largest user bases for Avocado in general, i.e. Avocado VT is a power user of sorts. For now the best I could suggest is gradually migrating crucial functionality that could be useful within Avocado itself to shrink the volume of this extra large plugin, then see where this gets us and what parts of its core we can salvage.